File: wheystore/wheystore.Web/Modules/SanPhamDB/DanhMucSanPham/DanhMucSanPhamRow.cs
namespace wheystore.SanPhamDB;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;

[ConnectionKey("Default"), Module("SanPhamDB"), TableName("DanhMucSanPham")]
[DisplayName("Danh Muc San Pham"), InstanceName("Danh Muc San Pham")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class DanhMucSanPhamRow : Row<DanhMucSanPhamRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Ma Loai"), Identity, IdProperty]
    public int? MaLoai { get => fields.MaLoai[this]; set => fields.MaLoai[this] = value; }

    [DisplayName("Ten Loai"), Size(100), NotNull, QuickSearch, NameProperty]
    public string TenLoai { get => fields.TenLoai[this]; set => fields.TenLoai[this] = value; }

    [DisplayName("Mo Ta")]
    public string MoTa { get => fields.MoTa[this]; set => fields.MoTa[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field MaLoai;
        public StringField TenLoai;
        public StringField MoTa;

    }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/DanhMucSanPhamRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface DanhMucSanPhamRow {
    MaLoai?: number;
    TenLoai?: string;
    MoTa?: string;
}

export abstract class DanhMucSanPhamRow {
    static readonly idProperty = 'MaLoai';
    static readonly nameProperty = 'TenLoai';
    static readonly localTextPrefix = 'SanPhamDB.DanhMucSanPham';
    static readonly lookupKey = 'SanPhamDB.DanhMucSanPham';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<DanhMucSanPhamRow>('SanPhamDB.DanhMucSanPham') }
    static async getLookupAsync() { return getLookupAsync<DanhMucSanPhamRow>('SanPhamDB.DanhMucSanPham') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<DanhMucSanPhamRow>();
}